### PR TITLE
feat(carousel): implementation according to ds5 and ds3

### PR DESCRIPTION
**Pull request by zettca**
Migrate `HvImageCarousel` component & move it to core widgets.

- Change component to receive `children` slides, instead of `documents`. Users can pass anything as _Slides_.
  - Renamed from `HvImageCarousel` to `HvCarousel`
  - Add `HvCarouselSlide` to use as `children`. Can take `img` props like `src`/`alt`, or `children` for custom content
- Add [embla-carousel](https://www.embla-carousel.com/)
  - adds drag&drop functionality and many [other options](https://www.embla-carousel.com/api/options/)
- Add `actions` area
- Improve docs
- Fix thumbnail opacity
- Default 16/9 image aspect-ratio
- Adds props to configure which Carousel elements are visible (title, dots, counter, thumbnails)
  - renames some props to opt-in more "boolean-ish" names

### CODE REVIEW

**Comment by MEsteves22** [comment]
While reviewing the PR, I noticed the following :
- Our internal classes should be listed before the classes provided by users in order for the external classes to overwrite the internal ones. For example: `cx(componentClasses.root, css(styles.root), className, classes.root)` .
- Add `HvCarousel` to the "Renamed components and types" section in the migration guide.
- Add `HvCarousel` to the "Promoted components" section in the migration guide.
- Remove "Image carousel" from the "UI Kit packages" section in the migration guide. It's listed there as a component missing from the lab package.

I really liked the optimised classes/styling logic!

File: packages/core/src/components/Carousel/CarouselSlide/CarouselSlide.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export const HvCarouselSlide = ({
+  classes = {},
+  children,
+  size: flexBasis = "100%",
+  src,
+  alt,
+  ...props
+}: HvCarouselSlideProps) => {
+  return (
+    <ClassNames>
```
Comment (MEsteves22) [inline packages/core/src/components/Carousel/CarouselSlide/CarouselSlide.tsx:29]: I believe you forgot to use the `makeClasses` util you've created here.

File: docs/overview/migration/MigrationV5.stories.mdx [hunk 2]
```diff
@@ -440,6 +440,7 @@ In v5.x the components below were promoted from the lab to the core package:
 - Time ago
 - Vertical navigation
 - Color picker
+- Carousel
```
[the remark this answers is not in the document]
Comment (zettca) [reply docs/overview/migration/MigrationV5.stories.mdx:443]: This is the "promoted to core" section. Am I missing something? 🤔

[the remark this answers is not in the document]
Comment (MEsteves22) [reply docs/overview/migration/MigrationV5.stories.mdx:443]: You are right, my bad!

**Review by MEsteves22 — APPROVED** [review]

**Review by francisco-guilherme — APPROVED** [review]

**Pull request merged**